### ISSUE TITLE
Rails/ReflectionClassName does not allow object converted to String with `to_s`

### ISSUE THREAD

**Issue by floehopper** [rubocop-hq/rubocop#7050]
In v0.69.0, the `Rails/ReflectionClassName` cop fails if the `class_name` of a Rails `belongs_to` association is set to an object which is converted to a String by calling the `#to_s` method on it. I don't see why this should not be allowed.

--------

## Expected behavior

`Rails/ReflectionClassName` cop passes.

## Actual behavior

`Rails/ReflectionClassName` cop fails.

## Steps to reproduce the problem

```ruby
class Comment < ApplicationRecord
  belongs_to :parent, class_name: model_name.to_s
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.69.0 (using Parser 2.6.3.0, running on ruby 2.5.5 x86_64-darwin17)
```

**Comment by tejasbubane** [comment]
Duplicate of https://github.com/rubocop-hq/rubocop/issues/6764. Please see comments there for the rationale.
